curriculum/challenges/english/25-front-end-development/quiz-css-attribute-selectors/66ed8fd0f45ce3ece4053eaf.md

```diff
@@ -17,439 +17,617 @@ Answer all of the questions below correctly to pass the quiz.
+Suppose, on freeCodeCamp’s forum, each discussion has a `data-status` attribute that could be `"open"` or `"closed"`. You want to highlight all open discussions. How would you select these discussions?
 
 #### --distractors--
 
-Placeholder distractor 1
-
+```css
+div[data-status$="open"] { border: 2px solid green; }
+```
 ---
 
-Placeholder distractor 2
+```css
+div[data-status^="open"] { border: 2px solid green; }
+```
 
 ---
 
-Placeholder distractor 3
+```css
+div[data-status*="open"] { border: 2px solid green; }
+```
 
 #### --answer--
 
-Placeholder answer
+```css
+div[data-status="open"] { border: 2px solid green; }
+```
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+ You want to style `<textarea>` elements in your form, but only those with a `maxlength` attribute. Which selector would work?
 
 #### --distractors--
 
-Placeholder distractor 1
+```css
+textarea[maxlength="true"] { border-color: darkred; }
+```
 
 ---
 
-Placeholder distractor 2
+```css
+textarea[maxlength^="true"] { border-color: darkred; }
+```
 
 ---
 
-Placeholder distractor 3
+```css
+textarea[maxlength*="true"] { border-color: darkred; }
+```
 
 #### --answer--
 
-Placeholder answer
+```css
+textarea[maxlength] { border-color: darkred; }
+```
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Suppose, on freeCodeCamp’s course catalog, each course has a `data-completion` attribute representing the percentage completed. You want to style courses where this attribute ends with `"100"`, indicating full completion. Which selector would work?
 
 #### --distractors--
 
-Placeholder distractor 1
+```css
+div[data-completion="100"] { font-weight: bold; }
+```
 
 ---
 
-Placeholder distractor 2
+```css
+div[data-completion*="100"] { font-weight: bold; }
+```
 
 ---
 
-Placeholder distractor 3
+```css
+div[data-completion^="100"] { font-weight: bold; }
+```
 
 #### --answer--
 
-Placeholder answer
+```css
+div[data-completion$="100"] { font-weight: bold; }
+```
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+You're building a job portal and want to style job listings where the `data-role` attribute contains `"developer"`. How would you write this selector?
 
 #### --distractors--
 
-Placeholder distractor 1
+```css
+div[data-role="developer"] { background-color: lightgreen; }
+```
 
 ---
 
-Placeholder distractor 2
+```css
+div[data-role^="developer"] { background-color: lightgreen; }
+```
 
 ---
 
-Placeholder distractor 3
+```css
+div[data-role$="developer"] { background-color: lightgreen; }
+```
 
 #### --answer--
 
-Placeholder answer
+```css
+div[data-role*="developer"] { background-color: lightgreen; }
+```
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+You want to style lists that have a type attribute with a value ending in `"square"`, indicating they use square bullets. Which selector would you use?
 
 #### --distractors--
 
-Placeholder distractor 1
+```css
+ul[type*="square"] { list-style-type: square; }
+```
+
```

Comment:
````suggestion
```css
ul[type*="square"] { list-style-type: square; }
```
````

let's not add empty lines that are not needed

Reply:
deleted the extra line now in commit: [8b705e3](https://github.com/freeCodeCamp/freeCodeCamp/pull/56611/commits/8b705e3e566119548c0f2c6937818a1fd3d0c447)